src/allmydata/immutable/offloaded.py

```diff
@@ -128,9 +139,9 @@ class CHKUploadHelper(Referenceable, upload.CHKUploader):
     peer selection, encoding, and share pushing. I read ciphertext from the
     remote AssistedUploader.
     """
-    VERSION = { "http://allmydata.org/tahoe/protocols/helper/chk-upload/v1" :
+    VERSION = { b"http://allmydata.org/tahoe/protocols/helper/chk-upload/v1" :
                  { },
-                "application-version": str(allmydata.__full_version__),
+                b"application-version": allmydata.__full_version__.encode("utf-8"),
```

Comment:
I don't know what this `str` was here for since `__full_version__` is already a str (ie, bytes) on Python 2 as far as I can tell.

The `.encode("utf-8")` seems fine and maybe it's required on Python 3 to make this bytes?  Since it is already a Python 2 str this is really an implicit ASCII decode followed by a UTF-8 encode which _could_ fail if the version has non-ASCII in it for some reason but it doesn't seem like it ever should.  I don't know if there's a better way to express "this string should be bytes on Python 2 and 3" and what's here.  If not, okay.  The extra decode will at least go away once Python 2 support is dropped.

Reply:
I based this off of prior art:

https://github.com/tahoe-lafs/tahoe-lafs/blob/574c63d3505f66fd49497d3bf05490d45fc0aa57/src/allmydata/immutable/offloaded.py#L506-L509